pom.xml

```diff
@@ -354,39 +323,40 @@
   </reporting>
 
   <profiles>
-
     <profile>
-      <id>publish</id>
-
-      <build>
-        <plugins>
-          <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-source-plugin</artifactId>
-            <executions>
-              <execution>
-                <id>attach-sources</id>
-                <goals>
-                  <goal>jar</goal>
-                </goals>
-              </execution>
-            </executions>
-          </plugin>
-
-          <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-javadoc-plugin</artifactId>
-            <executions>
-              <execution>
-                <id>attach-javadocs</id>
-                <goals>
-                  <goal>jar</goal>
-                </goals>
-              </execution>
-            </executions>
-          </plugin>
-        </plugins>
-      </build>
+      <id>default</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```

Comment:
Add this at the end of the `profiles` section:
```xml
+    <profile>
+      <id>testfail</id>
+      <properties>
+          <sampleprop>sdqs</sampleprop>
+      </properties>
+    </profile>
```

Then run `mvn validate -Ptestfail` and see. No modules at all.

The problem is not with how it works now. But how it will sooner or later fail when you or someone else adds a new profile to this POM.